test/deploySystem.ts

```diff
@@ -54,20 +55,24 @@ export async function deploySystem(config: RuntimeConfig, debug = false, useFall
     0,
     serviceRegistryAddress,
   ])
-  await uSwap.setPool(
-    '0xae7ab96520de3a18e5e111b5eaab095312d7fe84',
-    '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
-    10000,
-  )
-  await uSwap.addFeeTier(20)
+
+  await uSwap
+    .connect(signer)
+    .setPool(
+      '0xae7ab96520de3a18e5e111b5eaab095312d7fe84',
+      '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
+      10000,
+    )
+  await uSwap.connect(signer).addFeeTier(20)
 
   const [swap, swapAddress] = await deploy(CONTRACT_NAMES.common.SWAP, [
     address,
     ADDRESSES.main.feeRecipient,
     0,
     serviceRegistryAddress,
   ])
-  await swap.addFeeTier(20)
+
+  await swap.connect(signer).addFeeTier(20)
```

Comment:
Why do you explicitly connect the signer?

Reply:
Because, otherwise Hardhat is defaulting back to the default signer after resetting nodes. See -> helpers/restoreSnapshot.ts

After a node reset, we re-impersonate the rich account and have to connect the signer again.